utils/virtual_machines.py

```diff
@@ -34,8 +34,15 @@ def _vm_cleanup(mgmt, vm_name):
             'Could not destroy VM/instance {} ({}: {})'.format(vm_name, type(f).__name__, str(f)))
 
 
-def deploy_template(provider_key, vm_name, template_name=None, timeout=900, **deploy_args):
-
+def deploy_template(provider_key, vm_name, template_name=None, purpose=None,
+        timeout=900, **deploy_args):
```

Comment:
You should put the purpose either as a last positional argument (behind timeout) or inside kwargs. This could break older code that might not use names for the parameters.